### PR TITLE
Fix Error/Info/Warning Descriptors on TGraph/TH1/TMultiGraph

### PR DESCRIPTION
**Pull request by vargasa**
Fix Error/Info/Warning Descriptors on TGraph/TH1/TMultiGraph

### CODE REVIEW

**Comment by linev** [comment]
Does it required at all?
Error(), Info(), Warning() methods of TObject automatically add class name to the output.

    Error("TGraph", "Pointer to histogram is null");

will produce exactly the output:

    TGraph::TGraph() - Pointer to histogram is null

**Comment by Axel-Naumann** [comment]
Right, this will now print `TGraph::TGraph::TGraph()`.

Also for next time, please no merge commits but re-base.

@vargasa okay to close it or are we misunderstanding something?

File: hist/hist/src/TH1.cxx [hunk 1]
```diff
@@ -873,27 +873,27 @@ Bool_t TH1::Add(const TH1 *h1, Double_t c1)
...
       else
-         Warning("Add","Attempt to add histograms with different bin limits");
+         Warning("TH1::Add()","Attempt to add histograms with different bin limits");
    } catch(DifferentLabels&) {
       // in case of different labels -
       if (useMerge)
-         Info("Add","Attempt to add histograms with different labels - trying to use TH1::Merge");
+         Info("TH1::Add()","Attempt to add histograms with different labels - trying to use TH1::Merge");
       else
-         Info("Warning","Attempt to add histograms with different labels");
```
Comment (vargasa) [inline hist/hist/src/TH1.cxx]: Can we replace them by Warning() statements?

File: hist/hist/src/TH1.cxx [hunk 2]
```diff
@@ -5486,7 +5486,7 @@ Long64_t TH1::Merge(TCollection *li)
 Bool_t TH1::Multiply(TF1 *f1, Double_t c1)
 {
    if (!f1) {
-      Error("Add","Attempt to multiply by a non-existing function");
```
Comment (vargasa) [inline hist/hist/src/TH1.cxx]: This should be changed to Multiply. Right?

**Comment by vargasa** [comment]
@linev, If we are sure about that then I can just remove them. Last PR I did was to fix a similar issue (https://github.com/root-project/root/pull/1491) where I was getting just Get() without any reference to `TTreeReaderValue`. There are a couple of Info() statements being used to give Warnings. Should we replace them with Warning statements?. Finally, there are one or two statements in which the Descriptor must change. Please check the comments on the code.

**Comment by linev** [comment]
> Last PR I did was to fix a similar issue (#1491) where I was getting just Get() without any reference to TTreeReaderValue

This was different issue - at that place was subclass without dictionary. 
In such case class name may be misleading.

> There are a couple of Info() statements being used to give Warnings. Should we replace them with Warning statements?. 

Not sure - there are many places in ROOT where warnings produced with Info() method.
I would not change only one place because of that.

> Finally, there are one or two statements in which the Descriptor must change. 
> Please check the comments on the code.

I recommend to close with PR and open new one, where only wrong method descriptor should be changed.

File: hist/hist/src/TH1.cxx (hunk 2 above)
Comment (Axel-Naumann) [reply hist/hist/src/TH1.cxx]: Indeed :-)

File: hist/hist/src/TH1.cxx (hunk 1 above)
Comment (Axel-Naumann) [reply hist/hist/src/TH1.cxx]: `"Warning"` makes no sense here. I agree that this one should call `Warning()` not `Info()`. @lmoneta what do you think? Maybe even `Error()`?

**Comment by vargasa** [comment]
@linev Sure. Let's do that. Thanks guys!

File: hist/hist/src/TH1.cxx (hunk 1 above)
Comment (vargasa) [reply hist/hist/src/TH1.cxx]: Docs said this should be Location. But as @linev pointed out, there are some other places with using the same structure

**Pull request closed without merging**